### PR TITLE
*DDO-1585 Create CronJob in order to accept new connections in test environments

### PR DESCRIPTION
**Pull request by knapii-developments**
Restarts OpenDJ nightly at 10PM

Chart: 

```
# Source: opendj/templates/cronjob.yaml
apiVersion: batch/v1beta1
kind: CronJob
metadata:
  name: opendj-statefulset-restart
  labels:
    helm.sh/chart: "opendj-0.15.0"
    app.kubernetes.io/name: opendj
    app.kubernetes.io/instance: "opendj"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/component: opendj
    app.kubernetes.io/part-of: terra
spec:
  # If it's time for a new job run and the previous job run hasn't finished yet, the cron job skips the new job run
  concurrencyPolicy: Forbid
  schedule: '0 02 * * *' # Run once a day at 2 am UTC / 10 pm EST
  jobTemplate:
    metadata:
      labels:
        helm.sh/chart: "opendj-0.15.0"
        app.kubernetes.io/name: opendj
        app.kubernetes.io/instance: "opendj"
        app.kubernetes.io/managed-by: "Helm"
        app.kubernetes.io/component: opendj
        app.kubernetes.io/part-of: terra
    spec:
      backoffLimit: 2
      activeDeadlineSeconds: 600
      template:
        spec:
          serviceAccountName: opendj-service-sa
          # Prevents cronjobb from restarting on failure
          restartPolicy: Never
          containers:
            - name: kubectl
              image: bitnami/kubectl
              command:
                - 'kubectl'
                - 'rollout'
                - 'restart'
                - 'statefulset'
                - "opendj-statefulset"
```

### CODE REVIEW

**Review by jyang-broad — APPROVED** [review]
Looks Sane!

File: charts/opendj/templates/cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        spec:
+          serviceAccountName: {{ .Chart.Name }}-service-sa
+          # Prevents cronjobb from restarting on failure
+          restartPolicy: Never
+          containers:
+            - name: kubectl
+              image: bitnami/kubectl
+              command:
+                - 'kubectl'
+                - 'rollout'
```
Comment (mflinn-broad) [inline charts/opendj/templates/cronjob.yaml:26]: Do you need to specify the namespace opendj is in here in order for this to work?

File: charts/opendj/templates/cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ .Chart.Name }}-statefulset-restart
+  labels: {{ include "opendj.labels" . | nindent 4 }}
+spec:
+  # If it's time for a new job run and the previous job run hasn't finished yet, the cron job skips the new job run
+  concurrencyPolicy: Forbid
+  schedule: '0 02 * * *' # Run once a day at 2 am UTC / 10 pm EST
```
Comment (mflinn-broad) [inline charts/opendj/templates/cronjob.yaml:9]: No worries if you've already discussed this with chelsea, but does it need to be restarted everyday? The problem only occurs once per month.

Comment (knapii-developments) [reply charts/opendj/templates/cronjob.yaml:9]: That's what the [ticket](https://broadworkbench.atlassian.net/browse/DDO-1585?focusedCommentId=51775&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-51775) says as a proposed solution.  
>  A proposed solution is to add a CronJob that will restart the deployment nightly.

Comment (knapii-developments) [reply charts/opendj/templates/cronjob.yaml:9]: And in this [comment](https://broadworkbench.atlassian.net/browse/DDO-1585?focusedCommentId=51775). I would mention that there was no formal discussion. However, you do bring up a good point.

**Pull request merged**